build-local.rb

```diff
@@ -237,8 +245,75 @@ def generate_env(images)
   print "     Done\n"
 end
 
+def docker_ps
+  output = `docker ps --format '"{{$v1 := split .Image ":"}}{{$shortName := index $v1 0}}{{$shortName}}": { "name": "{{.Names}}", "image": "{{.Image}}", "container": "{{.ID}}" },'`
+  JSON.parse("{ #{output[0..-3]} }")
+end
+
+def list_compoents(repos)
```

Comment:
There's a few `compoents` that have slipped in so it might be worth running a find and replace and `s/compoents/components/g`

Reply:
I was trying to decide if that was the right name or if I should go for service?

Reply:
Either works for me. I think we have a kill-all-hub-services.sh in ida-hub-acceptance-tests if you wanted to be consistent with that.